File: Webapp/Front-end/src/app/endpoints/zaposlenik-endpoint/zaposlenik-uredi.endpoint.ts
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {MojConfig} from "../../moj-config";
import {Injectable} from "@angular/core";
@Injectable({providedIn: 'root'})
export class ZaposlenikUrediEndpoint implements MyBaseEndpoint<ZaposlenikUrediRequest, number>{
  constructor(public httpClient:HttpClient) {
  }
  obradi(request: ZaposlenikUrediRequest): Observable<number> {
    let url=MojConfig.adresa_servera+`/Zaposlenici/uredi/`+request.id;

    return this.httpClient.put<number>(url, request);
  }
}
export interface ZaposlenikUrediRequest {

  id: number,
  ime: string,
  prezime: string,
  ulica: string,
  brojMobitela: string,
  gradId: number

}

File: Webapp/Api/PC_Web_Shop/Endpoints/KorisnikEndpoints/ZaposlenikEndpoints/ZaposlenikUrediEndpoint/ZaposlenikUrediEndpoint.cs
using FIT_Api_Example.Helper;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using PC_Web_Shop.Data;
using PC_Web_Shop.Data.Models;
using PC_Web_Shop.Endpoints.KorisnikEndpoints.ZaposlenikEndpoints.ZaposlenikDodajEndpoint;
using PC_Web_Shop.Helper;
using PC_Web_Shop.Helper.Services;

namespace PC_Web_Shop.Endpoints.KorisnikEndpoints.ZaposlenikEndpoints.ZaposlenikUrediEndpoint
{
    [Route("Zaposlenici")]
    public class ZaposlenikUrediEndpoint:MyBaseEndpoint<ZaposlenikUrediRequest,ActionResult>
    {
        private readonly ApplicationDbContext _applicationDbContext;
        private readonly MyAuthService _myAuthService;

        public ZaposlenikUrediEndpoint(ApplicationDbContext applicationDbContext, MyAuthService authService)
        {
            _myAuthService = authService;
            _applicationDbContext = applicationDbContext;
        }
        [HttpPut("uredi/{Id}")]
        public override async Task<ActionResult> Obradi([FromBody]ZaposlenikUrediRequest request, CancellationToken cancellationToken)
        {
            if (!_myAuthService.IsLogiran())
            {
                return Unauthorized("Nije logiran");
            }
            var korisnickiNalog = _myAuthService.GetAuthInfo().KorisnickiNalog!;
            if (!korisnickiNalog.isAdmin)
            {

                return Unauthorized("Nije autorizovan");

            }

            var _odabraniZaposlenik = _applicationDbContext.Zaposlenik.Find(request.Id);
            if (_odabraniZaposlenik == null)
            {
                return BadRequest("Pogresan ID");
            }
            
            _odabraniZaposlenik.Ime = request.Ime;
            _odabraniZaposlenik.Prezime=request.Prezime;
            _odabraniZaposlenik.Ulica=request.Ulica;
            _odabraniZaposlenik.BrojMobitela = request.BrojMobitela;
            _odabraniZaposlenik.GradId=request.GradId;


            await _applicationDbContext.SaveChangesAsync(cancellationToken);
            return Ok(_odabraniZaposlenik.Id);
        }
    }
}
